Sources/LiveViewNative/Property Wrappers/LiveBinding.swift

```diff
@@ -0,0 +1,153 @@
+import LiveViewNativeCore
+import Combine
+
+/// Live bindings provide a mechanism for sharing state between the server and client in a way that can be updated by either.
+///
+/// Updates to the value from the client-side will automatically send an event to the server to update its stored value.
+/// Likewise, whenever the value is updated on the server, an event will be sent to the client which will trigger the `@LiveBinding` to update and the SwiftUI view that contains it (as well as any that use the ``projectedValue`` binding) to re-render.
+///
+/// ### Binding Names
+/// The name of a live binding is not defined directly by the client.
+/// Instead, it always controlled by the backend, to prevent the name getting out-of-sync, especially when multiple client versions may be in use.
+/// The name of the live binding is given by the value of the attribute whose name is provided to the ``LiveBinding/init(attribute:)`` initializer.
+///
+/// ### Server Support
+/// As live bindings are two-way mechanism, they are not implemented purely on the client, but rather require server-side support.
+///
+/// The server sends an event on mount with all of the binding values, sends an event when a binding is changed, and handles an even when a binding is changed on the client. These are all provided by the [`live_view_native_swift_ui`](https://github.com/liveviewnative/live_view_native_swift_ui) Elixir package.
+///
+/// The `bindings` macro takes a keyword list of bindings and their initial values and automatically generates all the necessary supporting code.
+/// The Elixir value used for the initial binding value must be serializable as JSON.
+///
+/// ```elixir
+/// defmodule AppWeb.TestLive do
+///     use AppWeb, :live_view
+///     use LiveViewNativeSwiftUi.Bindings
+///
+///     bindings(toggle_binding: false)
+/// end
+/// ```
+///
+/// Then, in the template, the same name is given as the value of a binding attribute:
+///
+/// ```html
+/// <my-toggle is-on="toggle_binding" />
+/// ```
+///
+/// ### Client Usage
+/// To use this property wrapper, the wrapped type must impelement the `Codable` protocol to define how values are serialized over the network.
+///
+/// ```swift
+/// struct MyToggle: View {
+///     @LiveBinding(attribute: "is-on") private var isOn: Bool
+///
+///     var body: some View {
+///         Toggle(isOn: $isOn) {
+///             Text("My Toggle")
+///         }
+///     }
+/// }
+/// ```
+@propertyWrapper
+public struct LiveBinding<Value: Codable> {
+    private let attributeName: AttributeName
+    @StateObject private var data = Data()
+    
+    @ObservedElement private var element
+    @Environment(\.coordinatorEnvironment) private var coordinator
+    @EnvironmentObject private var liveViewModel: LiveViewModel
+    
+    private var bindingName: String {
+        guard let name = element.attributeValue(for: attributeName) else {
+            fatalError("@LiveBinding missing binding name for \(attributeName)")
```

Comment:
If we were to use live bindings for things like `List.selection`, there would need to be a way to have the binding be optional. It's possible the user wouldn't use the selection. In that case, I think it should never change from the default value.